ksqldb-engine/src/main/java/io/confluent/ksql/physical/pull/operators/TableScanOperator.java

```diff
@@ -82,6 +87,10 @@ public Object next() {
           .get(nextLocation.getPartition());
     }
 
+    if (shouldCancelOperations.isDone()) {
```

Comment:
shouldn't this be the first thing that is checked?

Reply:
Sure.  For some reason, it seemed fine to setup the iterator, but that's really not required as you point out.